compiler/src/sct_checker_forward.ml

```diff
@@ -953,7 +961,12 @@ let move_msf ~loc env (msf, venv) mso msi =
 (* --------------------------------------------------------------- *)
 (* [ty_instr env msf i] return msf' such that env, msf |- i : msf' *)
 
-let rec ty_instr is_ct_asm fenv env ((msf,venv) as msf_e :msf_e) i =
+let rec ty_instr is_ct_asm fenv env (msf, venv) i =
+  let venv = Env.venv_forget (fst i.i_info) venv in
+  let msf, venv = ty_instr_r is_ct_asm fenv env (msf, venv) i in
+  msf, Env.venv_forget (snd i.i_info) venv
```

Comment:
Why forget is done after. I have the impression that it will be done again at the beginning of the next instruction. But maybe it is not really important.

Reply:
If you don’t do it, there are assertion failures in the test suite.

Sometimes, the next step is not the analysis of an instruction but a merge, e.g., in a `for` loop.

Reply:
Ok, let's merge it like that.